### PR TITLE
chore(fs): treat '/' as valid driver letter 

### PR DESCRIPTION
**Pull request by XuNeo**
When set `LV_FS_POSIX_LETTER = '/'`, do not generate warnings.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
Is using `LV_FS_DEFAULT_DRIVE_LETTER` an option for you?

**Comment by XuNeo** [comment]
Hi @kisvegabor ,

I'm not fully understand how to use `LV_FS_DEFAULT_DRIVE_LETTER`. The basic need is to set image in abs path like "/system/resource/image.bin"

**Comment by kisvegabor** [comment]
You can enable `LV_USE_FS_POSIX` and set
-  `LV_FS_POSIX_LETTER 'A'`
- `LV_FS_DEFAULT_DRIVE_LETTER 'A'`

This way when the driver letter is missing in the path, it will be redirected to posix (as the default letter shows posix).

**Comment by XuNeo** [comment]
That sounds not intuitive.
Set `LV_FS_POSIX_LETTER = '/' ` is more straight forward.

**Comment by kisvegabor** [comment]
However it won't work a relative path.

**Comment by XuNeo** [comment]
Hi @kisvegabor ,

Can we merge this? The definition LV_FS_POSIX_LETTER = '/' has worked well since LVGL v8 and provides a simple, effective solution. I don't see a reason to maintain the warning against its usage."

**Comment by kisvegabor** [comment]
What about this?
```c
#if LV_FS_POSIX_LETTER == '\0'
    #error "LV_FS_POSIX_LETTER must be set to an upper case letter"
#else
    /*Allow only upper case letters and '/'  ('/' is a special case for backward compatibility)*/
    #if (LV_FS_POSIX_LETTER != '/' && (LV_FS_POSIX_LETTER < 'A') || (LV_FS_POSIX_LETTER > 'Z'))
        #error "LV_FS_POSIX_LETTER must be an upper case ASCII letter"
    #endif
#endif
```

If you agree please update the other FS drivers too.

**Comment by liamHowatt** [comment]
When the default drive letter was added, we agreed to keep `'/'` as a valid drive letter. I think this warning was an accident.

**Comment by XuNeo** [comment]
Updated. Please help to review. Thanks!

**Review by FASTSHIFT — DISMISSED** [review]

**Review by kissa96 — APPROVED** [review]

File: src/libs/fsdrv/lv_fs_win32.c [hunk 1]
```diff
@@ -17,17 +17,9 @@
...
-    #if (LV_FS_WIN32_LETTER < 'A') || (LV_FS_WIN32_LETTER > 'Z')
-        #if LV_FS_DEFAULT_DRIVE_LETTER != '\0' /*When using default drive letter, strict format (X:) is mandatory*/
-            #error "LV_FS_WIN32_LETTER must be an upper case ASCII letter"
-        #else /*Lean rules for backward compatibility*/
-            #warning LV_FS_WIN32_LETTER should be an upper case ASCII letter. \
-            Using a slash symbol as drive letter should be replaced with LV_FS_DEFAULT_DRIVE_LETTER mechanism
-        #endif
-    #endif
+
+#if !LV_FS_IS_VALID_LETTER(LV_FS_STDIO_LETTER)
```
Comment (liamHowatt) [inline src/libs/fsdrv/lv_fs_win32.c]:
```suggestion
#if !LV_FS_IS_VALID_LETTER(LV_FS_WIN32_LETTER)
```

**Review by liamHowatt — CHANGES_REQUESTED** [review]
Looks good otherwise

File: src/libs/fsdrv/lv_fs_win32.c (hunk 1 above)
Comment (XuNeo) [reply src/libs/fsdrv/lv_fs_win32.c]: Nice catch. I double checked and still missed this one. :(

**Review by liamHowatt — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by andrewleech** [comment]
As a follow up to this, I've just found the default `LV_FS_POSIX_LETTER` in  [lv_conf_template.h](https://github.com/lvgl/lvgl/blob/abdcc04f09ef9c958d7f6cfc6579c16cbfe353e9/lv_conf_template.h#L796) is still `'\0'` so causes a compile fail if the template is used as-is.

Could this file get updated to match the new requirements here, I'm not too sure what the preferred character should be?

**Pull request merged**